rpcd/playbooks/roles/elasticsearch/files/reindex-liberty.py

```diff
@@ -0,0 +1,254 @@
+                                                           slave_index,
+                                                           master_count,
+                                                           slave_count,
+                                                           reindex_pct)
+        total_count += master_count
+        done_count += slave_count
+    try:
+        pct_done = (float(done_count) / float(total_count)) * 100
+    except ZeroDivisionError:
+        pct_done = float(0.00)
+    if parsed_args.verbose:
+        print("Total Docments: {0:20d} "
+              "Reindexed Documents: {1:20d} "
+              "Percentage complete: {2:3.2f}%").format(total_count,
+                                                       done_count,
+                                                       pct_done)
+    else:
+        print("Percent complete: {0:3.2f}%").format(pct_done)
+    if pct_done == 100:
+        exit(code=0)
+    else:
+        return False
+
+
+def drop_legacy(es, parsed_args):
+    """Drop the legacy logstash-* indices."""
+    for index in get_indices(es, parsed_args):
+        if parsed_args.suffix not in index and parsed_args.prefix in index:
+            if not parsed_args.dry_run:
+                es.indices.delete(index)
+            print("Dropping Legacy Index: {}").format(index)
+
+
+def list_indices(es):
+    """List all indices."""
+    print(es.cat.indices(v=True))
+
+
+def batch(es, es_host, parsed_args):
+    """Batch processing, useful for performing re-indexing of old indices
+    prior to the actual upgrade."""
+    today = date.today()
+    one_day = timedelta(days=1)
+    yesterday = today - one_day
+    indices = get_indices(es, parsed_args)
+    if not parsed_args.start:
+        d_start = None  # Date object for start date
+        p_start = yesterday  # Previous first day for sort
+        for index in indices:
+            if parsed_args.suffix not in index:
+                s_date = _index_to_date(index)
+                if s_date < p_start:
+                    p_start = s_date
+        d_start = p_start
+    elif parsed_args.start:
+        (s_year, s_month, s_day) = parsed_args.start.split('.')
+        d_start = date(int(s_year), int(s_month), int(s_day))
+    if not parsed_args.end:
+        d_end = yesterday
+    elif parsed_args.end:
+        (e_year, e_month, e_day) = parsed_args.end.split('.')
+        d_end = date(int(e_year), int(e_month), int(e_day))
+
+    numdays = d_end - d_start + one_day
+    date_range = [d_end - timedelta(days=x) for x in range(0, numdays.days)]
+
+    print("Start: {0}\tEnd: {1}\n").format(d_start, d_end)
+
+    for index in indices:
+        parsed_args.index = index
+        i_date = _index_to_date(index)
+        if i_date in date_range:
+            print("Re-indexing: {}").format(index)
+            if not parsed_args.dry_run:
+                reindex(es, es_host, parsed_args)
+
+
+def main():
+    parser = argparse.ArgumentParser()
```

Comment:
--host, --reindex, --monitor, --continuous, --drop appear to be the only CLI flags used. When will the others be used?